server/src/main/java/org/apache/uniffle/server/DefaultFlushEventHandler.java

```diff
@@ -153,7 +219,7 @@ protected Executor createFlushEventExecutor(int poolSize, String threadFactoryNa
 
   @Override
   public int getEventNumInFlush() {
-    return flushQueue.size();
+    return (int) ShuffleServerMetrics.gaugeEventQueueSize.get();
```

Comment:
Metrics may be unreliable.

Reply:
Could you give some proof? Although I also accept using a independent var to meet this requirement. And the original impl is not correct.

Reply:
I think twice. Currently, it is ok that using metric, because the function of `getEventNumInFlush` is to display this metric. Metric is easy to be overlooked, and it is easily be changed to incorrect values in the future.